js/components/dateRange.js

```diff
@@ -0,0 +1,280 @@
+Fliplet.FormBuilder.field('dateRange', {
+  name: 'Date range',
+  category: 'Text inputs',
+  props: {
+    value: {
+      type: Object,
+      default: {
+        start: '',
+        end: ''
+      }
+    },
+    description: {
+      type: String
+    },
+    autofill: {
+      type: String,
+      default: 'always'
+    },
+    defaultSource: {
+      type: String,
+      default: 'submission'
+    },
+    defaultEndDays: {
+      type: Number,
+      default: 2
+    },
+    showPredefinedRanges: {
+      type: Boolean,
+      default: true
+    },
+    placeholder: {
+      type: String,
+      default: function() {
+        return '-- ' + T('widgets.form.dateRange.rangePlaceholder');
+      }
+    }
+  },
+  data: function() {
+    return {
+      dateRange: null,
+      isInputFocused: false,
+      isPreview: Fliplet.Env.get('preview'),
+      today: moment().locale('en').format('YYYY-MM-DD'),
+      selectedRange: ''
+    };
+  },
+  mounted: function() {
+    this.initDaterange();
+
+    if (this.defaultValueSource !== 'default') {
+      this.setValueFromDefaultSettings({
+        source: this.defaultValueSource,
+        key: this.defaultValueKey
+      });
+    }
+
+    if (this.autofill === 'default' || this.autofill === 'always') {
+      this.value = {
+        start: this.today,
+        end: this.today
+      };
+      this.empty = false;
+    }
+
+    if (this.autofill === 'empty') {
+      this.value = {
+        start: '',
+        end: ''
+      };
+
+      return;
+    }
+
+    this.$emit('_input', this.name, this.value, false, true);
+    this.$v.$reset();
+  },
+  validations: function() {
+    var rules = {
+      value: {}
+    };
+
+    if (this.required && !this.readonly) {
+      rules.value.required = window.validators.required;
+    }
+
+    return rules;
+  },
+  computed: {
+    isApplyCurrentDateField: function() {
+      return this.autofill === 'always' || this.autofill === 'default';
+    },
+    predefinedRanges: function() {
```

Comment:
I don't think this needs to be a computed property. It should be fine as a `data` property.